### PR TITLE
fix(partition-plan): failed to generate partition name when using custom generator

### LINKED ISSUE #3132: [Bug]: failed to generate partition name on ob-mysql mode
### ODC version

4.3.1

### OB version

not related

### What happened?

1. create a table contains upper case partition key:
```sql
CREATE TABLE `datetime_range_parti_tbl` (
  `id` int(11) DEFAULT NULL,
  `parti_key` datetime DEFAULT NULL
) DEFAULT CHARSET = utf8mb4 ROW_FORMAT = DYNAMIC COMPRESSION = 'zstd_1.3.8' REPLICA_NUM = 1 BLOCK_SIZE = 16384 USE_BLOOM_FILTER = FALSE TABLET_SIZE = 134217728 PCTFREE = 0
 partition by range columns(PARTI_KEY)
(partition `p20200301` values less than ('2020-03-01 00:00:00'),
partition `p20210301` values less than ('2021-03-01 00:00:00'),
partition `p20220301` values less than ('2022-03-01 00:00:00'),
partition `p20230301` values less than ('2023-03-01 00:00:00'),
partition `p20240301` values less than ('2024-03-01 00:00:00'))
```

2. create a partition plan and use custom partition name generator:

截屏2024-08-16 20 35 52

as you can see, I ref the **UPPER CASE** partition key and I click 'test generate':
截屏2024-08-16 20 36 48

3. I change ref the lower case partition key and I click 'test generate':
截屏2024-08-16 20 37 48

### What did you expect to happen?

I can ref the UPPER CASE partition key

### How can we reproduce it (as minimally and precisely as possible)?

follow what happened

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by yhilmare**
#### What type of PR is this?

type-bug

#### What this PR does / why we need it:
failed to generate partition name when using custom generator
#### Which issue(s) this PR fixes:

Fixes #3132 #3133

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/plugins/task-plugin-ob-mysql/src/main/java/com/oceanbase/odc/plugin/task/obmysql/partitionplan/OBMySQLAutoPartitionExtensionPoint.java [hunk 1]
```diff
@@ -128,7 +128,7 @@ public boolean supports(@NonNull DBTablePartition partition) {
 
     @Override
     public String unquoteIdentifier(@NonNull String identifier) {
-        return StringUtils.unquoteMySqlIdentifier(identifier.toLowerCase());
+        return StringUtils.unquoteMySqlIdentifier(identifier);
```
Comment (guowl3) [inline server/plugins/task-plugin-ob-mysql/src/main/java/com/oceanbase/odc/plugin/task/obmysql/partitionplan/OBMySQLAutoPartitionExtensionPoint.java:131]: I have reopened issue 3016 and will resolve it in the next version.

**Review by guowl3 — APPROVED** [review]
lgtm

**Review by LuckyPickleZZ — DISMISSED** [review]

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/job/PartitionPlanJob.java [hunk 2]
```diff
@@ -129,6 +141,16 @@ public void execute(JobExecutionContext context) {
                     paramemters.getTimeoutMillis(), paramemters.getErrorStrategy());
         } catch (Exception e) {
             log.warn("Failed to execute a partition plan task", e);
+            AlarmUtils.alarm(AlarmEventNames.PARTITION_PLAN_SCHEDULED_FAILED, e);
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/job/PartitionPlanJob.java:144]: not stable, we cannot garantee no exception thrown here, we may enforce logic in AlarmUtils.alarm

File: server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/job/PartitionPlanJob.java [hunk 3]
```diff
@@ -129,6 +141,16 @@ public void execute(JobExecutionContext context) {
...
+            if (this.notificationProperties.isEnabled()) {
+                try {
+                    Event event = this.eventBuilder.ofFailedTask(this.flowInstanceService
+                            .getTaskByFlowInstanceId(paramemters.getFlowInstanceId()));
+                    this.broker.enqueueEvent(event);
+                } catch (Exception ex) {
+                    log.warn("Failed to enqueue event.", ex);
```
Comment (yizhouxw) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/schedule/job/PartitionPlanJob.java:151]: log output context should contains event info

**Review by yizhouxw — APPROVED** [review]

**Review by LuckyPickleZZ — APPROVED** [review]

**Review by guowl3 — APPROVED** [review]

**Pull request merged**